packages/react/ds-app-launchpad/src/ui/MarkdownEditor/MarkdownEditor.tsx

```diff
@@ -0,0 +1,188 @@
+    >
+      <div className="top-bar">
+        {!hidePreview && (
+          <ViewModeTabs
+            editMode={editMode}
+            onEditModeChange={handleEditModeChange}
+          />
+        )}
+        {!hideToolbar && (
+          <Toolbar label="Markdown Editor">
+            <ToolbarGroup label="Text Formatting">
+              <ToolbarButton
+                label={toolbar.title.label}
+                onClick={toolbar.title.handler}
+                shortcut={toolbar.title.shortcut}
+              >
+                {icons.ToolbarTitle}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.bold.label}
+                onClick={toolbar.bold.handler}
+                shortcut={toolbar.bold.shortcut}
+              >
+                {icons.ToolbarBold}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.italic.label}
+                onClick={toolbar.italic.handler}
+                shortcut={toolbar.italic.shortcut}
+              >
+                {icons.ToolbarItalic}
+              </ToolbarButton>
+            </ToolbarGroup>
+            <ToolbarSeparator />
+            <ToolbarGroup label="Tools">
+              <ToolbarButton
+                label={toolbar.quote.label}
+                onClick={toolbar.quote.handler}
+                shortcut={toolbar.quote.shortcut}
+              >
+                {icons.ToolbarQuote}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.code.label}
+                onClick={toolbar.code.handler}
+                shortcut={toolbar.code.shortcut}
+              >
+                {icons.ToolbarCode}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.link.label}
+                onClick={toolbar.link.handler}
+                shortcut={toolbar.link.shortcut}
+              >
+                {icons.ToolbarLink}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.uList.label}
+                onClick={toolbar.uList.handler}
+                shortcut={toolbar.uList.shortcut}
+              >
+                {icons.ToolbarUnorderedList}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.oList.label}
+                onClick={toolbar.oList.handler}
+                shortcut={toolbar.oList.shortcut}
+              >
+                {icons.ToolbarOrderedList}
+              </ToolbarButton>
+            </ToolbarGroup>
+          </Toolbar>
+        )}
+      </div>
+
+      <textarea
+        defaultValue={defaultValue}
+        placeholder={placeholder}
+        className="editor-content"
+        ref={textareaRef}
+        style={style}
+        hidden={editMode !== "write"}
+      />
+      {editMode === "preview" && (
+        <div className="editor-content" style={style} ref={previewRef}>
+          <ReactMarkdown
```

Comment:
This markdown editor, as configured, is missing a nice QOL feature: when editing a list (unordered or ordered), and pressing enter, I'd expect another list item to be created beneath the one I was just editing. Is it possible to add that with some configuration option?

Reply:
Added some autocompletion. It is

Reply:
Thanks for adding - yeah, I agree this level of maturity is good enough for an initial version.